packages/lit-dev-tools-cjs/src/playground-plugin/blocking-renderer.ts

```diff
@@ -82,7 +138,44 @@ export class BlockingRenderer {
     });
   }
 
+  private getCachedRender(lang: string, code: string): string | null {
+    if (!this.isDevMode) {
+      return null;
+    }
+    const fileName = createUniqueFileNameKey(lang, code);
+    const absoluteFilePath = pathlib.resolve(cachedHighlightsDir, fileName);
+    if (fs.existsSync(absoluteFilePath)) {
+      return fs.readFileSync(absoluteFilePath, {encoding: 'utf8'});
+    }
+    return null;
+  }
+
+  private writeCachedRender(lang: string, code: string, html: string) {
+    const fileName = createUniqueFileNameKey(lang, code);
+    const absoluteFilePath = pathlib.resolve(cachedHighlightsDir, fileName);
+    fs.writeFileSync(absoluteFilePath, html);
+  }
+
   render(lang: 'js' | 'ts' | 'html' | 'css', code: string): {html: string} {
+    if (!this.isDevMode) {
+      // In production, skip all caching.
+      return this.renderWithWorker(lang, code);
+    }
+    // In dev mode, speed up the edit-refresh loop by caching the syntax
+    // highlighted code.
+    const cachedResult = this.getCachedRender(lang, code);
```

Comment:
Nit: Can we create unique filename key above this, and have both `getCachedRender()` and `writeCachedRender()` accept the key instead of lang & code so we don't need to redo the hash?

Reply:
Yes! Excellent performance based catch! definitely will do that.

Reply:
Done